src/api/java/jni/cvc5_Grammar.cpp

```diff
@@ -0,0 +1,120 @@
+#include "cvc5_Grammar.h"
+
+#include "api/cpp/cvc5.h"
+#include "cvc5JavaApi.h"
+
+using namespace cvc5::api;
+
+/*
+ * Class:     cvc5_Grammar
+ * Method:    deletePointer
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_cvc5_Grammar_deletePointer(JNIEnv*,
+                                                       jclass,
+                                                       jlong pointer)
+{
+  delete ((Grammar*)pointer);
+}
+
+/*
+ * Class:     cvc5_Grammar
+ * Method:    addRule
+ * Signature: (JJJ)V
+ */
+JNIEXPORT void JNICALL Java_cvc5_Grammar_addRule(JNIEnv* env,
+                                                 jobject,
+                                                 jlong pointer,
+                                                 jlong ntSymbolPointer,
+                                                 jlong rulePointer)
+{
+  CVC5_JAVA_API_TRY_CATCH_BEGIN;
+  Grammar* current = (Grammar*)pointer;
+  Term* ntSymbol = (Term*)ntSymbolPointer;
+  Term* rule = (Term*)rulePointer;
+  current->addRule(*ntSymbol, *rule);
+  CVC5_JAVA_API_TRY_CATCH_END(env);
+}
+
+/*
+ * Class:     cvc5_Grammar
+ * Method:    addRules
+ * Signature: (JJ[J)V
+ */
+JNIEXPORT void JNICALL Java_cvc5_Grammar_addRules(JNIEnv* env,
+                                                  jobject,
+                                                  jlong pointer,
+                                                  jlong ntSymbolPointer,
+                                                  jlongArray rulePointers)
+{
+  CVC5_JAVA_API_TRY_CATCH_BEGIN;
+  Grammar* current = (Grammar*)pointer;
+  Term* ntSymbol = (Term*)ntSymbolPointer;
+  // get the size of pointers
+  jsize size = env->GetArrayLength(rulePointers);
+  // allocate buffer for the long array
+  jlong* cRules = new jlong[size];
+  // copy java array to the buffer
+  env->GetLongArrayRegion(rulePointers, 0, size, cRules);
+  // copy the terms into a vector
+  std::vector<Term> rules;
+  for (jsize i = 0; i < size; i++)
+  {
+    Term* term = (Term*)cRules[i];
+    rules.push_back(*term);
+  }
+  // free the buffer memory
+  delete[] cRules;
```

Comment:
```suggestion
  std::vector<jlong*> cRules(size);
  // copy java array to the buffer
  env->GetLongArrayRegion(rulePointers, 0, size, cRules.data());
  // copy the terms into a vector
  std::vector<Term> rules;
  for (jlong* cRule : cRules)
  {
    Term* term = static_cast<Term*>(cRule);
    rules.push_back(*term);
  }
```

Reply:
It may be a good idea to have a generic helper function for converting a `jlongArray` to an `std::vector<T>`.

Reply:
Done. Thanks for the suggestion. I missed that I can use data here. This pattern is common, so I will refactor others later. 
static_cast has the same issue.

Reply:
Refactored using function `getObjectsFromPointers`  declared [here](https://github.com/cvc5/cvc5/pull/6396/files#diff-253cd17df3736f0725828a64c58967b3b587b12c83b680c48ffaf9f663d90e30R50).